### ISSUE TITLE
Move godot4 branch into master

### ISSUE THREAD

**Issue by nxuv** [godot-dlang/godot-d#15]
It's more of a proposal, but maybe it'll be better to move master into godot3, godot4 into master and remove gd4 & releases, since currently project is being worked with intent of porting to godot 4, it'd make more sense

**Comment by nxuv** [comment]
Also maybe detach a fork and move into own repo?
With respectful mention of original project

**Comment by Superbelko** [comment]
Nah, I'm not going to step in as a maintainer for that, and without that there is no reason keep this as a fork.

**Comment by nxuv** [comment]
I wish there was a way to contact godot-d account maintainer, coz in godot-d repo issues are certainly not watched

**Comment by nxuv** [comment]
Still, how about reorganising branches?

**Comment by Superbelko** [comment]
gd4 is from original repo, not sure if I can delete it here. (**edit**: I can)

before moving godot4 branch into a master I need to google up if it can mess something when there is similar branch changes in original repo or something.

**Comment by nxuv** [comment]
Well, I just did that and I think it might screw some fork syncing
But on the other hand I don't think sheepandshepherd would continue working on godot-d

**Comment by nxuv** [comment]
I might offer a solution, as which: I could create new org and step in as maintainer, plus I'd make you member so you could maintain it in free time
Basically do what sheepandshepherd did, but not dead

**Comment by nxuv** [comment]
Reasoning in that is tbh quite simple
I have a lot of free time, never liked gdscript, kinda dislike c#, too oop imho
So for me maintaining it might be good solution

**Comment by Superbelko** [comment]
Let's returns to that topic in 2 weeks. 

But before you start, do you really need godot-d? 
I mean, I initially started it because I was excited by such potentially progressive and slim alternative to Unity and UE4, then in the spring I went back and implement most of the current features because I needed a back plan in case of potential licensing restrictions for UE5, currently... well if you look up tutorial on youtube most of the cool features are implemented as plugins and until Godot 4 full release there isn't much to do, and my main work is not related to gamedev. That being said, despite how much I don't like C++ and UE5 I do understand that the tools it offers simply out of reach by godot.

So, it is cool and all but simply unrealistic for me to finish anything but smallest games, but after many failures with different teams I completely lost interest in that.

As for the free time, it took me a whole year to update it into something meaningful, while the real effort was only like somewhere between 200-300 hours.

**Comment by nxuv** [comment]
> But before you start, do you really need godot-d?

Honestly, yes. I do need it. I really had enough of c# and it doesn't feel good on large scale. It's sole purpose is being MS attempt of getting Java's pie. And GDScript's syntax feels so alien to me I physically can't code with it for more then fixing someone's plugins

> tools it offers simply out of reach by godot.

imho I don't mind it being like that. Godot's features are perfect for me aside couple of hurdles and some editor stuff. I'm not aiming at AAA type of games and just want to have my nice lang work with my nice game engine

> So, it is cool and all but simply unrealistic for me to finish anything but smallest games, but after many failures with different teams I completely lost interest in that.

I get that. Been there

> As for the free time, it took me a whole year to update it into something meaningful, while the real effort was only like somewhere between 200-300 hours.

I'm absolutely not in rush and I actually spend most of my time coding, might as well code something usefull

I was actually just too afraid to jump into godot-d coz it was a mess of a project, I wasn't able to even set up a minimal project with that and all in all seeing that it's still alive by your efforts lighted up fire in me once more

**Comment by nxuv** [comment]
Well, if you'd want to jump in then I made project [here](https://github.com/godot-dlang/godot-dlang)

**Comment by Superbelko** [comment]
I think I could just transfer ownership of this repo to keep already opened issues and comments. 
What do you think about it?

**upd**: so it looks like I can simply join your organization, and transfer all issues instead as described here
https://docs.github.com/en/issues/tracking-your-work-with-issues/transferring-an-issue-to-another-repository

**Comment by nxuv** [comment]
Sounds good, I believe I've already invited you. Gonna send another one tho

**Comment by Superbelko** [comment]
Ok, joined. 
I think you now need to set up roles and access in that organization so I can push/commit etc...

**Comment by nxuv** [comment]
Should be done now.
Avoid any pushes to repo for now, I'm doing heavy reorganising

**Comment by Superbelko** [comment]
Yeah, but it seems I should transfer it to you or organization first.

**Comment by Superbelko** [comment]
Done. I have already transfered first issue, the rest is up to you. 

I suggest keep this fork here so people can find out what is happening, just update readme with explanations and directions to look for a new repo.

(**edit**) The good thing is that it seems links does gets redirected into this organization repo and everything still works. yay \0/

From now I will go silent for few days.

**Comment by ParchmentWolf** [comment]
Im interested to help out in any way possible :O --likewise I dislike C# and the support for it in godot adds bloat as big as the engine itself lol.

I would argue I, as well as a good few friends NEED godot with D support... Rust isnt any fun, and C/C++ doesnt come with any quality of life when coding in it... D gives us the power AND a fun time coding. I have a project I've been trying to rework into godot as it solves alot of issues I had when trying to make it in C# initially (before I abandoned that trainwreck of a lang) which does emulator/modding stuff and would attract more users to DLang and Godot in general.
